packages-ts/starknet-gauntlet-ledger/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import { IStarknetWallet, Env } from '@chainlink/starknet-gauntlet'
+import { SignerInterface, encode } from 'starknet'
+import { Abi, Signature, Invocation, InvocationsSignerDetails } from 'starknet/types'
+import { TypedData, getMessageHash } from 'starknet/utils/typedData'
+import { fromCallsToExecuteCalldataWithNonce } from 'starknet/utils/transaction'
+import { calculcateTransactionHash, getSelectorFromName } from 'starknet/utils/hash'
+import Stark, { LedgerError } from '@ledgerhq/hw-app-starknet'
+
+// EIP-2645 path
+//  2645 - EIP number
+//  579218131 - layer - 31 lowest bits of sha256("starkex")
+//  894929996 - application - 31 lowest bits of sha256("chainlink")
+export const DEFAULT_LEDGER_PATH = "m/2645'/579218131'/894929996'/0'"
+export const LEDGER_PATH_REGEX = /^\s*m\s*\/\s*2645\s*\'\s*\/\s*579218131\s*\'\s*\/\s*(\d+)\s*\'\s*\/\s*(\d+)\s*\'$/
+
+class LedgerSigner implements SignerInterface {
+  client: Stark
+  path: string
+  publicKey: string
+
+  private constructor(client: Stark, path: string, publicKey: string) {
+    this.client = client
+    this.path = path
+    this.publicKey = publicKey
+  }
+
+  static create = async (path?: string): Promise<LedgerSigner> => {
+    if (!path) {
+      path = DEFAULT_LEDGER_PATH
+    } else {
+      const match = LEDGER_PATH_REGEX.exec(path)
+      if (!match) {
+        throw new Error(
+          "Provided ledger path doesn't correspond the expected format m/2645'/579218131'/<application>'/<index>'",
+        )
+      }
+    }
+
+    // work around jest reimporting the package
+    // package is only allowed to be imported once
+    const TransportNodeHid = (await import('@ledgerhq/hw-transport-node-hid')).default
```

Comment:
Not sure I understand the problem here?
What's different [than on Solana](https://github.com/smartcontractkit/chainlink-solana/blob/571a53a9eecc44fd8bec60522ec098a0f01512ff/gauntlet/packages/gauntlet-solana/src/commands/wallet.ts#L3)?

Reply:
Jest compiler complains here for some reason. I dig into it at some point, and this was the best solution I could find. 
Solana doesn't have tests, so we don't have this problem...

Reply:
Jest attempts to import the package multiple times when compiling. The package only allows to be imported once (it takes ownership of the socket when imported).